internal/service/advancedcluster/resource_advanced_cluster_test.go

```diff
@@ -788,6 +846,34 @@ func configSingleProvider(projectID, name string) string {
 	`, projectID, name)
 }
 
+func configIncorrectTypeGobalClusterSelfManagedSharding(projectID, name string) string {
+	return fmt.Sprintf(`
+		resource "mongodbatlas_advanced_cluster" "test" {
+			project_id   = %[1]q
+			name         = %[2]q
+
+			cluster_type = "REPLICASET"
+			global_cluster_self_managed_sharding = true
```

Comment:
nit: could we add a small comment here why this config is incorrect?

Reply:
thx, comment added